test/interop/circuit/index.js

```diff
@@ -0,0 +1,209 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+const parallel = require('async/parallel')
+const series = require('async/series')
+const Factory = require('../../utils/ipfs-factory-daemon')
+
+const crypto = require('crypto')
+const utils = require('./utils')
+
+const multiaddr = require('multiaddr')
+
+chai.use(dirtyChai)
+
+describe('circuit interop', function () {
+  let jsTCP
+  let jsTCPAddrs
+  let jsWS
+  let jsWSAddrs
+  let jsRelayAddrs
+  let factory = new Factory()
+
+  let goRelayAddrs
+  let goRelayDaemon
+
+  let goTCPAddrs
+  let goTCPDaemon
+  let goTCP
+
+  let goWSAddrs
+  let goWSDaemon
+  let goWS
+
+  beforeEach((done) => {
+    parallel([
+      (pCb) => utils.setupJsNode([
```

Comment:
Please require this util above directly and use it as a single func.

Is there a way that setupJsNode can be replaced by ipfs-factory?